contracts/VestAMM/libraries/AmmIntegration/CurveVestAMM.sol

```diff
@@ -173,4 +153,108 @@ contract CurveVestAMM {
 
         return curvePool.get_dy(0, 1, 1 ether);
     }
+
+    ////////////////////////////////////
+    // Virtual Price Estimation Logic //
+    ////////////////////////////////////
+
+    /**
+     * @dev A snapshot of a liquidity pool at a given time - perhaps integrate into Struct used above?
+     */
+    struct Pooldata {
+        uint256 amountA; // Raw amounts of each token in the pool - assuming only two tokens in the pool
+        uint256 amountB;
+        uint256 ratioA; // (amountA * 1e18) / amountB
+        uint256 ratioB; // (amountB * 1e18) / amountA
+        uint256 lpTotal; // Total supply of LP Tokens
+        uint256 lpAmount; // Amount of LP tokens owned
+    }
+
+    Pooldata public firstPoolData;
+    Pooldata public secondPoolData;
+
+    function getPoolData(
+        IERC20 _tokenA,
+        IERC20 _tokenB,
+        IERC20 _lpContract,
+        IERC20 _lpToken
+    ) internal returns (Pooldata memory) {

```

Comment:
This also can be changed to:
`getPoolData(IVestAMM.VAmmInfo calldata _vammInfo)`